pyluba/bluetooth/ble_message.py

```diff
@@ -1252,3 +1233,156 @@ def close_clear_connect_current_wifi2(self, ssid: str, get_msg_cmd: int):
+        if not expect:
+            build = new_builder.set_type_value(test_id).set_time_of_duration(test_duration).build()
+        else:
+            try:
+                json_array = json.loads(expect)
+                z2 = True
+                for i in range(len(json_array)):
+                    new_builder2 = mctrl_sys_pb2.QCAppTestExcept.Builder()
+                    json_object = json_array[i]
+                    if "except_type" in json_object:
+                        string = json_object["except_type"]
+                        if "conditions" in json_object:
+                            json_array2 = json_object["conditions"]
+                            for i2 in range(len(json_array2)):
+                                json_object2 = json_array2[i2]
+                                new_builder3 = mctrl_sys_pb2.QCAppTestConditions.Builder()
+                                if "cond_type" in json_object2:
+                                    new_builder3.set_cond_type(json_object2["cond_type"])
+                                else:
+                                    z2 = False
+                                if "value" in json_object2:
+                                    obj = json_object2["value"]
+                                    if string == "int":
+                                        new_builder3.set_int_val(int(obj))
+                                    elif string == "float":
+                                        new_builder3.set_float_val(float(obj))
+                                    elif string == "double":
+                                        new_builder3.set_double_val(float(obj))
+                                    elif string == "string":
+                                        new_builder3.set_string_val(str(obj))
+                                    else:
+                                        z2 = False
+                                    new_builder2.add_conditions(new_builder3)
+                                else:
+                                    z2 = False
+                        new_builder2.set_except_type(string)
+                        new_builder.add_except(new_builder2)
+                        new_builder2.clear()
+                z = z2
+            except json.JSONDecodeError:
+                z = False
+            if z:
+                build = new_builder.set_type_value(test_id).set_time_of_duration(test_duration).build()
+            else:
+                build = new_builder.set_type_value(test_id).set_time_of_duration(test_duration).build()
+        print(f"Factory tool print, mow_to_app_qctools_info_t={build.except_count}, mow_to_app_qctools_info_t22={build.except_list}")
+        build2 = mctrl_sys_pb2.MctlSys(mow_to_app_qctools_info=build)
+        print(f"Send command - factory tool test command testId={test_id}, testDuration={test_duration}", "Factory tool print222", True)
+        self.send_order_msg_sys(build2)
+
+    def send_sys_set_date_time(self):
```

Comment:
also would be good to add comments python style beneath the function.
e.g """Update date and time on mower. """